tests/unit/scene/test_ipynb_conversion.py

```diff
@@ -0,0 +1,118 @@
+"""Unit tests for the conversion from and to ipynb."""
+
+from pytest_mock import MockerFixture
+import pytest_check as check
+import json
+
+from opencodeblocks.scene.from_ipynb_conversion import ipynb_to_ipyg, is_title
+from opencodeblocks.scene.ipynb_conversion_constants import BLOCK_TYPE_TO_NAME
+
+
+class TestIpynbConversion:
+
+    """Testing function_parsing functions"""
+
+    def test_empty_data(self, mocker: MockerFixture):
+        """Empty data should return empty document"""
+        check.equal(ipynb_to_ipyg({}), {"blocks": [], "edges": []})
+
+    def test_empty_notebook_data(self, mocker: MockerFixture):
+        """The conversion for an empty notebook should be coherent"""
+        file_path = "./tests/assets/empty.ipynb"
+        real_notebook_is_coherent(file_path)
+
+    def test_complex_notebook_data(self, mocker: MockerFixture):
+        """The conversion for a usual notebook should be coherent"""
+        file_path = "./tests/assets/usual.ipynb"
+        real_notebook_is_coherent(file_path)
+
+    def test_complex_notebook_data(self, mocker: MockerFixture):
+        """The conversion for a complex notebook should be coherent"""
+        file_path = "./tests/assets/complex.ipynb"
+        real_notebook_is_coherent(file_path)
+
+    def test_is_title(self, mocker: MockerFixture):
+        """Should return True iff the given text can be used as a title for a block"""
+        check.equal(is_title(string_to_markdown_block("")), False)
+        check.equal(is_title(string_to_markdown_block("Data Preprocessing")), True)
+        check.equal(is_title(string_to_markdown_block("Étude de cas")), True)
+        check.equal(is_title(string_to_markdown_block("# Report")), False)
+        check.equal(
+            is_title(
+                string_to_markdown_block(
+                    "This is a very very very very very very very very very very very very very very very very very very long explanation"
+                )
+            ),
+            False,
+        )
+        check.equal(is_title(string_to_markdown_block("New line \n Next line")), False)
+
+
+def real_notebook_is_coherent(file_path: str):
+    """
+    From ipynb conversion on a real notebooks should return
+    1. blocks and edges
+    2. the right amount of code blocks and edges
+    3. blocks and sockets with unique ids
+    4. edges with existing ids
+    5. code blocks that always have a source
+    6. markdown blocks that always have text
+    """
+    ipynb_data = load_json(file_path)
+    ipyg_data = ipynb_to_ipyg(ipynb_data)
+
+    # 1.
```

Comment:
Use text instead of numbers.
```suggestion
    # Blocks and edges
```